python/pyabacus/tests/test_diago_dav_subspace.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+import pytest
+from pyabacus import hsolver
+import numpy as np
+import scipy
+
+def diag_pyabacus(h_sparse, nband):
+    def mm_op(x):
+        return h_sparse.dot(x)
+
+    nbasis = h_sparse.shape[0]
+
+    v0 = np.random.rand(nbasis, nband)
+
+    diag_elem = h_sparse.diagonal()
+    diag_elem = np.where(np.abs(diag_elem) < 1e-5, 1e-5, diag_elem)
+    precond = 1.0 / np.abs(diag_elem)
+
+    e, _ = hsolver.dav_subspace(
+        mm_op,
+        v0,
+        nbasis,
+        nband,
+        precond,
+        dav_ndim=8,
+        tol=1e-4,
+        max_iter=5000,
+        scf_type=False
+    )
+    
+    return e
+
+def diag_eigsh(h_sparse, nband):
+    e, _ = scipy.sparse.linalg.eigsh(h_sparse, k=nband, which='SA', maxiter=5000, tol=1e-4)
+    return e
```

Comment:
It seems you set the stopping criterion to `1e-4`. I don't think it is a good stopping criterion since the threshold of SCF is at least 1e-7. If possible, try the `1e-7` and machine precision.

Reply:
i have changed it.